core/src/main/kotlin/com/malinskiy/marathon/analytics/internal/sub/ExecutionReport.kt

```diff
@@ -42,6 +48,39 @@ data class ExecutionReport(
             .flatten()
     }
 
+    private fun createTestSummaries(): Map<Test, TestSummary> {
+        val resultsByTest: Map<Test, List<TestResult>> = testEvents
+            .groupBy { it.testResult.test }
```

Comment:
groupBy(keySelector = { it.testResult.test }, valueTransform = { it.testResult }